### DISCUSSION TITLE
Search disappears when overriding base.html

### DISCUSSION THREAD

**Discussion by jonasbirkelof** [squidfunk/mkdocs-material#4311] [Q&A] [answered]
When I override base.html to insert a static .css before main.css, the search field disapperas from navigation. Anyone knows why? Do I have to add something to mkdocs.yml och some asset?

**Comment by squidfunk** [comment]
You very likely have either removed something related to search, or the whole application source code. Check the console for errors. When overriding the base template, it's a good idea to override specific blocks. Also make sure to call `{{ super () }}`, if you override the `scripts` block. Related #4309

**Reply by jonasbirkelof** [reply]
All I really did was adding "custom_dir: override" to mkdocs.yml, creating the override folder and creating a base.html inside it. I copied the contents from this file: https://github.com/squidfunk/mkdocs-material/blob/master/src/base.html.

In the console the `/javascripts/bundle.js` is not found. Where can I find it?

Maby it's overkill doing this? All I want is to insert a custom .css file before main.css.

**Comment by jonasbirkelof** [comment]
Well, I found out that I should use main.html and extend base.html. How it works but I still can't add a `<link>` before the main.css `<link>`. The extrahead block adds it below...

**Reply by squidfunk** [reply]
[Additional CSS](https://squidfunk.github.io/mkdocs-material/customization/#additional-css) is a much more resilient way to add CSS, as you don't even need to extend the theme. To add a `link` tag, the `extrahead` block can be used, but as you noted, it comes after all CSS has been defined. Thus, if you need to move it before the stylesheet, your only option is to extend `extrahead` and add the `link` tag, as well as your CSS there. An example how to override `extrahead` can be found [in our documentation](https://squidfunk.github.io/mkdocs-material/reference/?h=extrahead#on-all-pages).

**Comment by jonasbirkelof** [comment, marked as answer]
I don't know if it's the right approach, but adding the link using site_meta worked!
